website/src/components/Chat/ChatConversationTree.tsx

```diff
@@ -0,0 +1,104 @@
+import { Box } from "@chakra-ui/react";
+import { ChevronLeft, ChevronRight } from "lucide-react";
+import { memo, useCallback, useMemo, useState } from "react";
+import { InferenceMessage } from "src/types/Chat";
+import { buildTree, Tree } from "src/utils/buildTree";
+import { StrictOmit } from "ts-essentials";
+import { useIsomorphicLayoutEffect } from "usehooks-ts";
+
+import { BaseMessageEmojiButton } from "../Messages/MessageEmojiButton";
+import { MessageInlineEmojiRow } from "../Messages/MessageInlineEmojiRow";
+import { ChatMessageEntry, ChatMessageEntryProps } from "./ChatMessageEntry";
+
+type ChatConversationTreeProps = { messages: InferenceMessage[]; retryingParentId: string | null } & StrictOmit<
+  ChatMessageEntryProps,
+  "message"
+>;
+
+export const ChatConversationTree = memo(function ChatConversationTree({
+  messages,
+  ...props
+}: ChatConversationTreeProps) {
+  const tree = useMemo(() => buildTree(messages), [messages]);
+  if (!tree) return null;
+
+  return (
+    <>
+      <ChatMessageEntry message={tree} {...props}></ChatMessageEntry>
+      {tree.children.length > 0 && <TreeChildren trees={tree.children} {...props} />}
+    </>
+  );
+});
+
+const TreeChildren = ({
+  trees,
```

Comment:
I think the name here might be a bit confusing, this is not an array of trees, but one tree, correct?

Reply:
This is an array of trees, that why I called it `children`. But anyway if you have any better name please suggest.